### ISSUE TITLE
Improve Future Monad resolver

### ISSUE THREAD

**Issue by fvilante** [fvilante/NextRobot#2]
_Future Monad_ is used with a `resolver` function.

Would be better to call this resolver func like this: (proposed)
```typescript
const a = Future( (ok, reject) => ok(10) )
```
Instead of this: (current use)
```typescript
const a = Future( resolver => resolver(Right(10))
```

**Comment by fvilante** [comment]
//TODO IMPORTANT: Make management of Future state.
    //      IF CLIENT CALL OK OR ERROR MORE THAN ONE TIME, NOTHING MUST HAPPEN. FUTURE IS ALREADY FULLFILED.
    //      AVOID DOUBLE-ACTION!

**Comment by fvilante** [comment]
CLOSED in favor of #37
